### PR TITLE
[Feat]: Add Badge Component

### PR DESCRIPTION
**Pull request by AbhinavMV**
Things done in this PR:
1. Added Badge component with 3 props - text, state, textVariant (textVariant takes all values-h1,h2...etc that typography component can use)
2. Create stories for the component.
3. Written simple tests for the badge component.
Please find below screenshot of all the badges.
image

Please let me know your suggestions. 😄

### CODE REVIEW

File: src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { BadgeProps } from './types';
+
+const Badge: React.FC<BadgeProps> = ({
+    text,
+    state = 'normal',
+    textVariant = 'body16',
+    ...otherProps
+}) => {
+    return (
+        <DivStyled state={state} data-testid="test-badge-id" {...otherProps}>
```
Comment (locothedev) [inline src/components/Badge/Badge.tsx:13]: We do not want to allow other props to be passed

File: src/components/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import { Typography } from '../Typography';
+import { DivStyled } from './Badge.styles';
+import { BadgeProps } from './types';
+
+const Badge: React.FC<BadgeProps> = ({
+    text,
+    state = 'normal',
+    textVariant = 'body16',
```
Comment (locothedev) [inline src/components/Badge/Badge.tsx:9]: sort alphabetically for @BillyG83 's eyes 👀

File: src/components/Badge/types.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    /**
+     * Variant of text style
+     */
+    textVariant?: variantType;
+
+    /**
+     * Set the initial state for the Badge component
+     */
+    state?: colorState;
```
Comment (locothedev) [inline src/components/Badge/types.ts:19]: sort alphabetically please

**Review by locothedev — CHANGES_REQUESTED** [review]

File: src/components/Badge/Badge.tsx (hunk 1 above)
Comment (AbhinavMV) [reply src/components/Badge/Badge.tsx]: Removed it.

File: src/components/Badge/Badge.tsx (hunk 2 above)
Comment (AbhinavMV) [reply src/components/Badge/Badge.tsx]: Done.

File: src/components/Badge/types.ts (hunk 3 above)
Comment (AbhinavMV) [reply src/components/Badge/types.ts]: Done.

File: src/components/Badge/Badge.tsx (hunk 2 above)
Comment (BillyG83) [reply src/components/Badge/Badge.tsx]: there is a VSC plugin called sortLine @AbhinavMV TY dude!

File: src/components/Badge/Badge.styles.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            return color.yellow;
+        default:
+            return color.blue;
+    }
+};
+
+export const DivStyled = styled.div<TStyleProps>`
+    ${resetCSS}
+    ${fonts.text}
+    width: min-content;
```
Comment (BillyG83) [inline src/components/Badge/Badge.styles.ts:25]: `width` to `justify-content` can you sort alphabetically please.

We try to keep all props and CSS attribute in alphabetical sorted. But you nailed it with the imports first and the StyledComponent function last, bravo!

**Review by BillyG83 — CHANGES_REQUESTED** [review]
🤩 wow bravo sir. If you can fix that one tiny comment of mine in the styles file then I will merge you. Hit me up with your Twitter handle please so I can pamp you up!

**Comment by AbhinavMV** [comment]
> star_struck wow bravo sir. If you can fix that one tiny comment of mine in the styles file then I will merge you. Hit me up with your Twitter handle please so I can pamp you up!

Thanks for letting me work on this. It was a great learning experience.
Here is my twitter handle - @abhinav_mv5

**Review by locothedev — APPROVED** [review]

**Review by locothedev — APPROVED** [review]

**Pull request merged**